src/admin.py

```diff
@@ -133,6 +136,54 @@ def change_password(
     db.commit()
     print(f"Password updated for user '{username}'.")
 
+@app.command()
+def create_api_key(
+    username: str = typer.Argument(None, help="User to create API key for."),
+    key_name: str = typer.Option(
+        None, "--key-name", "-n", help="Name for the API key."),
+    key_description: Optional[str] = typer.Option(
+        None, "--description", "-d", help="Description for the API key (optional)."
+    ),
+):
+    """Create an API key for a user."""
+    db = database.SessionLocal()
+
+    # Check for existing user *before* potentially prompting
+    if not username and get_user_by_username_from_db(db, username):
```

Comment:
You sure the 'not' in this statement does what you want it to do? Without the 'not' it would trigger if the username is not None as a CLI argument AND it exists in the db, which seems to be what you want.

Reply:
Good catch - this check actually don't make sense here, I removed it.